### PR TITLE
Update Clientlib

### PR DESCRIPTION
**Pull request by DutChen18**
- [Add default type parameters to class templates](https://github.com/leaningtech/ts2cpp/commit/75e89b54d47c88211af48694268af80958c6a6fc)
- [Add template constructor helpers](https://github.com/leaningtech/ts2cpp/commit/a48e9e5d65165dc79d7bd966a863c5caec32891c)

Should be no breaking changes.

### CODE REVIEW

File: include/client/new/types.h [hunk 1]
```diff
@@ -437,12 +463,17 @@ namespace [[cheerp::genericjs]] client {
...
+		[[gnu::always_inline]]
+		static Function* _New(_Args... args) {
+			return new Function(args...);
+		}
 		static Function* prototype;
 	protected:
 		[[cheerp::client_transparent]]
 		Function(EventListener* listener) noexcept;
 		[[cheerp::client_transparent]]
-		Function(const EventListener* listener) noexcept;
```
Comment (yuri91) [inline include/client/new/types.h:445]: is dropping of noexcept intended?

**Review by yuri91 — COMMENTED** [review]
clientlib.h gets 14% bigger with this.
I see the reason for always adding the _New methods, but how big would the increase bit if they are only added where necessary?

File: include/client/new/types.h (hunk 1 above)
Comment (DutChen18) [reply include/client/new/types.h]: No, it was not intended.

This was the result of a mistake in an earlier commit that leaves me wondering how the `noexcept` was ever there in the first place.
image

Regardless, it's fixed now.

**Comment by DutChen18** [comment]
> clientlib.h gets 14% bigger with this. I see the reason for always adding the _New methods, but how big would the increase be if they are only added where necessary?

Despite the larger file size, there does not seem to be a significant increase in build times.

Only adding the `_New` methods for template constructors increases the size by just 0.3%.

I figure it's worth it anyways so there is a single consistent way to construct objects that always works, instead of making an exception for cases when it's needed. The consistency of `_New` could be useful for template metaprogramming.

What do you think?

**Comment by yuri91** [comment]
Indeed the consistency is valuable, so let's go ahead and add it everywhere

**Pull request merged**